apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.html.ts

```diff
@@ -0,0 +1,335 @@
+import { html, joinHtml, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+import { compiledScriptTag, nodeModulesAssetPath } from '../../lib/assets.js';
+import { config } from '../../lib/config.js';
+import { encodePath } from '../../lib/uri-util.js';
+
+export function InstructorFileEditor({ resLocals }: { resLocals: Record<string, any> }) {
+  const { fileEdit, authz_data, course_owners, course, __csrf_token } = resLocals;
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head data-ace-base-path="${nodeModulesAssetPath('ace-builds/src-min-noconflict/')}">
+        ${renderEjs(import.meta.url, "<%- include('../partials/head'); %>", {
+          ...resLocals,
+          pageTitle: `Edit ${fileEdit?.fileName}`,
```

Comment:
When would `fileEdit` be nullish? Could you pass `fileEdit` as a separate property outside of `resLocals`, or are you saving that for a future PR?

Reply:
> When would `fileEdit` be nullish?

When the user has no permission or this is the example course.

> Could you pass `fileEdit` as a separate property outside of `resLocals`, or are you saving that for a future PR?

I'm saving that for a future PR that properly handles the involved types (similar to #10224).

Reply:
Actually I'll likely get rid of `fileEdit` in favour of individual parameters for the actual data.